Throw explicit error when plugin configuration is missing

TylerHendrickson opened the pull request:
Hello,

I encountered an issue that surfaces when this plugin is listed in the Serverless `plugins` array but no `custom.customDomain` object is defined: a `Type Error` is thrown with the message `"Cannot read property 'enabled' of undefined"`. This error is thrown because there is no check to ensure that `this.serverless.service.custom.customDomain` exists before setting the `enabled` constant on line 78 of `index.js` (within the `ServerlessCustomDomain.evaluateEnabled()` method. While I would agree that if a plugin is enabled in Serverless it should be properly configured, it would be nice if this plugin handled such a case a little more gracefully.

Therefore, this pull request tweaks the current behavior so that instead of the default `Type Error` being thrown in this situation, it throws an error with the message `'serverless-domain-manager: Plugin configuration is missing.'`

captainsidd commented:
Hi there! This seems really useful, thank you for the PR. If you could add a unit test to catch this error as well as update the version and CHANGELOG, that'd be great. 

Thanks again!

TylerHendrickson commented:
Hi again! I added the items you requested (unit tests, version bump, and updated CHANGELOG). I also added an additional check (along with an associated test) to ensure that the `custom` object in Serverless exists, simply so that behavior is consistent.

Let me know if there's anything else!

captainsidd commented:
Hey there! Everything looks good. One tiny nitpick: seems as though this would be a patch rather than a minor version bump. If you could change that, we'll get this merged. 

Thanks for taking care of this!

captainsidd commented:
Perfect! Let's get this merged.
